pkg/klusterlet/clusterinfo/clusterinfo_controller.go

```diff
@@ -567,6 +567,77 @@ func (r *ClusterInfoReconciler) getDistributionInfoAndClusterID() (clusterv1beta
 	return distributionInfo, clusterID, nil
 }
 
+var ocpInfrGVR = schema.GroupVersionResource{
+	Group:    "config.openshift.io",
+	Version:  "v1",
+	Resource: "infrastructures",
+}
+
+const (
+	AWSPlatformType = "AWS"
+	GCPPlatformType = "GCP"
+	GCEPlatformType = "GCE"
+)
+
+func (r *ClusterInfoReconciler) getClusterRegion() string {
+	var region = ""
+
+	switch {
+	case r.isOpenshift():
+		obj, err := r.ManagedClusterDynamicClient.Resource(ocpInfrGVR).Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			klog.Errorf("failed to get OCP infrastructures.config.openshift.io/cluster: %v", err)
+			return ""
+		}
+
+		platformType, _, err := unstructured.NestedString(obj.Object, "status", "platform")
+		if err != nil {
+			klog.Errorf("failed to get OCP platform type in status of infrastructures.config.openshift.io/cluster: %v", err)
+			return ""
+		}
+
+		// only ocp on aws and gcp has region definition
+		// refer to https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go
+		switch platformType {
+		case AWSPlatformType:
+			region, _, err = unstructured.NestedString(obj.Object, "status", "platformStatus", "aws", "region")
+			if err != nil {
+				klog.Errorf("failed to get OCP region in status of infrastructures.config.openshift.io/cluster: %v", err)
+				return ""
+			}
+		case GCPPlatformType:
+			region, _, err = unstructured.NestedString(obj.Object, "status", "platformStatus", "gcp", "region")
+			if err != nil {
+				klog.Errorf("failed to get OCP region in status of infrastructures.config.openshift.io/cluster: %v", err)
+				return ""
+			}
+		}
+	}
+
+	return region
+}
+
+func (r *ClusterInfoReconciler) getInfraConfig() string {
+	if !r.isOpenshift() {
+		return ""
+	}
+
+	obj, err := r.ManagedClusterDynamicClient.Resource(ocpInfrGVR).Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("failed to get OCP infrastructures.config.openshift.io/cluster: %v", err)
+		return ""
+	}
+
+	infraName, _, err := unstructured.NestedString(obj.Object, "status", "infrastructureName")
+	if err != nil {
+		klog.Errorf("failed to get OCP infrastructure Name in status of infrastructures.config.openshift.io/cluster: %v", err)
+		return ""
+	}
+
+	infraConfig := fmt.Sprintf("{\"infraName\":\"%s\"}", infraName)
```

Comment:
it is better to have a struct for this and use json.Marshal

Reply:
done
```
type InfraConfig struct {
	InfraName string `json:"infraName,omitempty"`
}
```